mesonbuild/mtest.py

```diff
@@ -1855,17 +1856,34 @@ def tests_from_args(self, tests: T.List[TestSerialisation]) -> T.Generator[TestS
         run all tests with that name across all subprojects, which is
         identical to "meson test foo1"
         '''
+        patterns = {}
         for arg in self.options.args:
             if ':' in arg:
                 subproj, name = arg.split(':', maxsplit=1)
+                if name == '':
+                    name = '*'
+                if subproj == '':  # in case arg was ':'
+                    subproj = '*'
```

Comment:
So `foo:` is short for `foo:*` and `:` short for `*:*`.  I would personally make them syntax error, but why not... I don't think you documented that behaviour, did you?

Reply:
Note that this changes `foo` from meaning main project's test `foo`, to test `foo` from all subprojects. I don't think we have a syntax any more for specifying only main project, do we?

Reply:
The current syntax is that `foo` and `:foo` means all tests named `foo` in all projects, `bar:` means all tests in `bar` subproject, `bar:foo` means `foo` test of `bar` subproject, and `:` means all tests.  I don't think there is a way to run tests from main project only.
I only added the possibility to use `*` to specify partial names.

But, yes, I should update the documentation.

Reply:
>  I don't think there is a way to run tests from main project only.

You can... the main project's "subproject" tag is the same one it would be if it were a subproject so you just run tests for `mainprj:`